hub/integrations/pytorch_old.py

```diff
@@ -35,8 +41,18 @@ def __init__(
 
         self.dataset = dataset
         self.transform = transform
+        self.keys: List[str]
+        if tuple_fields is None:
+            self.keys = list(dataset.tensors)
+            self.tuple_mode = False
+        else:
+            for field in tuple_fields:  # type: ignore
+                if field not in dataset.tensors:
+                    raise TensorDoesNotExistError(field)
+            self.keys = tuple_fields
+            self.tuple_mode = True
```

Comment:
why is this a field variable? just have tuple fields be the field variable and if it is `None` you can assume it is not in tuple mode

Reply:
See @AbhinavTuli 's comment above^.

```python
I think we can simplify this slightly and remove self.tuple_fields by only using self.keys and assigning tuple_fields to to it if it is passed. This will also improve other functions like load_all_index_meta and load_all_meta as they won't fetch extra data for keys that are not needed.
```